Source/Core/VideoBackends/D3D/D3DState.cpp

```diff
@@ -259,7 +259,10 @@ ID3D11SamplerState* StateCache::Get(SamplerState state)
 
 	unsigned int mip = d3dMipFilters[state.min_filter & 3];
 
-	if (state.max_anisotropy > 1)
+	// Only use anisotropic filtering if one or both of the filters are set to Linear.
+	// If both filters are set to Point then using anisotropy is equivalent
+	// to "forced filtering" which will cause visual glitches.
+	if (state.max_anisotropy > 1 && (state.min_filter & 4 || state.mag_filter))
```

Comment:
`(state.min_filter & 4 || state.mag_filter)` should likely be extracted to its own descriptive template function in VideoCommon, considering it's shared across all three backends (perhaps in a new SamplerCommon.h or something)